File: server/Models/Manufacturer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace InsightWorks.Models;

public class Manufacturer
{
    [Key]
    [Column("id")]
    public Guid Id { get; set; } = Guid.NewGuid();
    
    [Required]
    [MaxLength(50)]
    [Column("manufacturer_code")]
    public string ManufacturerCode { get; set; } = null!;
    
    [Required]
    [MaxLength(100)]
    [Column("name")]
    public string Name { get; set; } = null!;
    
    [MaxLength(200)]
    [Column("address")]
    public string? Address { get; set; }
    
    [MaxLength(50)]
    [Column("contact_person")]
    public string? ContactPerson { get; set; }
    
    [MaxLength(20)]
    [Column("contact_phone")]
    public string? ContactPhone { get; set; }
    
    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    
    // 导航属性
    [JsonIgnore]
    public virtual ICollection<Equipment>? Equipments { get; set; }
} 

File: web/src/core/entities/manufacturer.ts
import dayjs from "dayjs";
import type { Manufacturer } from "@/typings/api";

export class ManufacturerEntity implements Manufacturer {
  id?: string;
  manufacturerCode: string;
  name: string;
  address?: string | null;
  contactPerson?: string | null;
  contactPhone?: string | null;
  createdAt?: string;
  updatedAt?: string;

  constructor(data: Partial<Manufacturer>) {
    Object.assign(this, data);
  }

  get formattedCreatedAt() {
    return this.createdAt
      ? dayjs(this.createdAt).format("YYYY-MM-DD HH:mm:ss")
      : "";
  }

  get formattedUpdatedAt() {
    return this.updatedAt
      ? dayjs(this.updatedAt).format("YYYY-MM-DD HH:mm:ss")
      : "";
  }
}
